### PR TITLE
feat: ability to edit badges

### LINKED ISSUE #420: ability to edit badges
**Describe the feature you'd like**
Creating a new activity which handles editing badges

**Would you like to work on the issue?**
YES

### PR DESCRIPTION
**Pull request by adityastic**
Fixes #420

Changes:
- Ability to Edit Badges

Screenshots for the change:

### CODE REVIEW

File: app/src/main/java/org/fossasia/badgemagic/ui/EditClipartActivity.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package org.fossasia.badgemagic.ui
+
+import android.os.AsyncTask
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.Observer
+import kotlinx.android.synthetic.main.activity_edit_badge.*
```
Comment (yashk2000) [inline app/src/main/java/org/fossasia/badgemagic/ui/EditClipartActivity.kt]: Is a wildcard import needed here?

Comment (adityastic) [reply app/src/main/java/org/fossasia/badgemagic/ui/EditClipartActivity.kt]: Draft PR xD. Still WIP

Comment (adityastic) [reply app/src/main/java/org/fossasia/badgemagic/ui/EditClipartActivity.kt]: got it xD

**Comment by adityastic** [comment]
@yashk2000 run it locally on your device

**Comment by yashk2000** [comment]
@adityastic on it.

**Comment by yashk2000** [comment]
@adityastic tested it. When I am pressing the edit button, the  app is crashing on my device.

**Comment by yashk2000** [comment]
`
2019-07-08 20:59:13.460 23107-23107/org.fossasia.badgemagic E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.fossasia.badgemagic, PID: 23107
    com.squareup.moshi.JsonDataException: Expected BEGIN_OBJECT but was BEGIN_ARRAY at path $
        at com.squareup.moshi.JsonUtf8Reader.beginObject(JsonUtf8Reader.java:162)
        at org.fossasia.badgemagic.data.fragments.BadgeConfigJsonAdapter.fromJson(BadgeConfigJsonAdapter.kt:42)
        at org.fossasia.badgemagic.data.fragments.BadgeConfigJsonAdapter.fromJson(BadgeConfigJsonAdapter.kt:17)
        at com.squareup.moshi.JsonAdapter$2.fromJson(JsonAdapter.java:137)
        at com.squareup.moshi.JsonAdapter.fromJson(JsonAdapter.java:41)
        at org.fossasia.badgemagic.util.SendingUtils.getBadgeFromJSON$app_debug(SendingUtils.kt:98)
        at org.fossasia.badgemagic.bindings.EditBadgeViewBindingsKt.setBadgeValues(EditBadgeViewBindings.kt:19)
        at org.fossasia.badgemagic.databinding.ActivityEditBadgeBindingImpl.executeBindings(ActivityEditBadgeBindingImpl.java:255)
        at androidx.databinding.ViewDataBinding.executeBindingsInternal(ViewDataBinding.java:472)
        at androidx.databinding.ViewDataBinding.executePendingBindings(ViewDataBinding.java:444)
        at androidx.databinding.ViewDataBinding$7.run(ViewDataBinding.java:196)
        at androidx.databinding.ViewDataBinding$6.onViewAttachedToWindow(ViewDataBinding.java:164)
        at android.view.View.dispatchAttachedToWindow(View.java:18401)
        at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3403)
        at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3410)
        at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3410)
        at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3410)
        at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3410)
        at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3410)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1771)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1469)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7209)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1050)
        at android.view.Choreographer.doCallbacks(Choreographer.java:862)
        at android.view.Choreographer.doFrame(Choreographer.java:797)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1036)
        at android.os.Handler.handleCallback(Handler.java:874)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:198)
        at android.app.ActivityThread.main(ActivityThread.java:6729)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
`

File: app/src/main/java/org/fossasia/badgemagic/ui/EditClipartActivity.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package org.fossasia.badgemagic.ui
+
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.Observer
+import kotlinx.android.synthetic.main.activity_edit_clipart.*
```
Comment (yashk2000) [inline app/src/main/java/org/fossasia/badgemagic/ui/EditClipartActivity.kt]: Wildcard import xD

Comment (adityastic) [reply app/src/main/java/org/fossasia/badgemagic/ui/EditClipartActivity.kt]: its kotlin  synthetic xD. It is allowed

Comment (yashk2000) [reply app/src/main/java/org/fossasia/badgemagic/ui/EditClipartActivity.kt]: Ohk :+1:

**Pull request merged**